runners/spark/src/main/java/org/apache/beam/runners/spark/translation/streaming/SparkRunnerStreamingContextFactory.java

```diff
@@ -52,54 +53,59 @@
   private final CheckpointDir checkpointDir;
 
   public SparkRunnerStreamingContextFactory(
-      Pipeline pipeline,
-      SparkPipelineOptions options,
-      CheckpointDir checkpointDir) {
+      Pipeline pipeline, SparkPipelineOptions options, CheckpointDir checkpointDir) {
     this.pipeline = pipeline;
     this.options = options;
     this.checkpointDir = checkpointDir;
   }
 
-  private EvaluationContext ctxt;
+  public Function0<JavaStreamingContext> create() {
```

Comment:
Can this be replaced by a `public static` member? It is already lazy thanks to `Function0#call`.